### PR TITLE
(Issue #518) SVG to PDF conversion code

### PR DESCRIPTION
**Pull request by jmcioffi4**
Issue: https://github.com/digitalcredentials/learner-credential-wallet/issues/518

Note:
The top value to fill in (CredentialSubject.name) seems off center but we have applied no styling changes so I am not sure why this occurs.

<br/>

Screenshot 2023-10-25 at 12 41 24 PM

<br/>

Screenshot 2023-10-25 at 12 41 36 PM

<br/>

Screenshot 2023-10-25 at 12 41 57 PM

### CODE REVIEW

**Comment by jmcioffi4** [comment]
@kayaelle @dmitrizagidulin we need to talk about the dynamic part (handlebars) in our next meeting before we can account for that

File: app/types/credential.ts [hunk 1]
```diff
@@ -82,6 +82,13 @@ export type Proof = {
   jws?: string;
 }
 
+export type RenderMethod = {
+  id: string;
+  type: string;
+  name: string;
```
Comment (dmitrizagidulin) [inline app/types/credential.ts:88]:
```suggestion
  id?: string;
  type: string;
  name?: string;
```

File: app/types/credential.ts [hunk 2]
```diff
@@ -82,6 +82,13 @@ export type Proof = {
...
+  css3MediaQuery: string;
```
Comment (dmitrizagidulin) [inline app/types/credential.ts:89]:
```suggestion
  css3MediaQuery?: string;
```

File: app/screens/PublicLinkScreen/PublicLinkScreen.tsx [hunk 3]
```diff
@@ -49,6 +52,45 @@ export default function PublicLinkScreen ({ navigation, route }: PublicLinkScree
     });
   }
 
+
+  const handleShareAsPdf = async() => {
+    // templateURL = svg template from id of renderMethod in Credential
+    const templateURL = credential.renderMethod?.[0].id; // might want to sort if there are more than one renderMethod
```
Comment (dmitrizagidulin) [inline app/screens/PublicLinkScreen/PublicLinkScreen.tsx:58]:
```suggestion
    // short-circuit exit if no render method
    if(!credential['renderMethod']) {
       return;
    }
    const templateURL = credential.renderMethod?.[0].id; // might want to sort if there are more than one renderMethod
```

File: android/settings.gradle [hunk 4]
```diff
@@ -1,4 +1,6 @@
 rootProject.name = 'Learner Credential Wallet'
+include ':react-native-html-to-pdf'
+project(':react-native-html-to-pdf').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-html-to-pdf/android')
```
Comment (dmitrizagidulin) [inline android/settings.gradle:3]: what is this line for?

File: app/screens/PublicLinkScreen/PublicLinkScreen.tsx [hunk 5]
```diff
@@ -49,6 +52,45 @@ export default function PublicLinkScreen ({ navigation, route }: PublicLinkScree
...
+    let source = '';
+    if (templateURL) {
+      await fetch(templateURL)
+        .then(response => {
+          return response.text();
+        }).then(data => {
+          source = data;
+        }).catch(error => {
+          console.error('An error occurred fetching the SVG Template:', error);
+        });
```
Comment (dmitrizagidulin) [inline app/screens/PublicLinkScreen/PublicLinkScreen.tsx:68]: Ok, so, `fetch()` is treacherous. You'd THINK that if there's an HTTP error (like 404 Not Found), the `.catch()` at the end would catch it. But not so :) You have to mess with `response.ok` instead. 
So the correct version of this line is something like:

```suggestion
      let source;
      try {
        const response = await fetch(templateURL);
        if (!response.ok) {
          throw new Error(`HTTP Error: ${response.status}`);
        }
        source = await response.json();
      } catch (e) {
        console.log('Error fetching template:', e);
      }
```

**Comment by jmcioffi4** [comment]
App properly converts valid handlebars into the pdf:

Screenshot 2023-10-30 at 3 22 01 PM

**Comment by jmcioffi4** [comment]
@dmitrizagidulin @kayaelle ready for review.

NOTE: The logic was moved to `app/lib/svgToPdf.ts` and we will move this into a separate repo sometime this week :)

**Comment by dmitrizagidulin** [comment]
Conflicts resolved, merging and releasing for testing.

**Pull request merged**